Add linters and formatters

gunnnnii opened the pull request:
This PR introduces ESLint, Stylelint and Prettier for linting and formatting.

It uses the standard ESLint config (generated with `npm init @eslint/config`) with the addition of `eslint-config-prettier`.

gunnnnii committed:
adds lint and formatting deps and config files

gunnnnii commented on src/main.ts:
```diff
@@ -29,24 +29,24 @@ const map = new WebScene({
 });
 
-map.when().then(() => {
+void map.when().then(() => {
```
Alternatively we can add an ignore or a `.catch` to these promises.

Semantically `void` is probably the cleanest thing to do but it's also a pretty unusual syntax so it might be better to do one of the other options...

arnofiva replied:
I like `void` and it also makes sense as a linter rule. As pointed out in the comment however, it would be good to have them reported as errors instead of warnings.

arnofiva commented:
Thanks, this was overdue!

A few questions:
* I can essentially reduce the `package.json` dependencies to the following and everything seems to work. Do you see a reason to keep `eslint-plugin-import`/`eslint-plugin-n`/`eslint-plugin-promise`/`stylelint-config-standard`? 
```
 "dependencies": {
    "@arcgis/core": "^4.27",
    "@esri/calcite-components": "1.4.2"
  },
  "devDependencies": {
    "@types/sortablejs": "^1.15.1",
    "@typescript-eslint/eslint-plugin": "^6.7.4",
    "eslint": "^8.50.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-config-standard-with-typescript": "^39.1.0",
    "npm-run-all": "^4.1.5",
    "prettier": "3.0.3",
    "sass": "^1.63.4",
    "stylelint": "^15.10.3",
    "stylelint-config-standard-scss": "^11.0.0",
    "typescript": "^5.2.2",
    "vite": "^4.3.9"
  }
```
* Can you ease the stylelint to allow camel case and revert the `viewDiv`? We use this in so many places and also the public samples. I believe this rule would cause more annoyance than appreciation.
* Can we change the linter messages in VS Code to [warnings instead of errors](https://stackoverflow.com/questions/64947493/change-all-eslint-errors-to-warning-instead-of-error-using-eslintrc-js)? 
* In the Readme.md, should we point out the required VS Code extensions to use linting? Btw, I get eslint errors but not stylelint, even though I have the extension installed. Can you check on your side?

gunnnnii committed:
disables the `selector-class-pattern` stylelint rule

gunnnnii committed:
- drops unnecessary `eslint-plugin` peer deps.
- uses `concurrently` instead of `npm-run-all` to have nicer output when running scripts in parallel

gunnnnii commented:
Could the fact that `stylelint-config-standard` works for you be the vscode plugin using a default config? Stylelint runs completely separate from eslint so this dep has nothing to do with it. I can remove the dep though if we don't want to worry about linting css files.

Are you seeing the errors in the editor or when you run the lint script? It could be that your editor is selecting a default config instead of what's defined in the workspace, can you confirm by running `npm i && npm run lint`. I'm neither getting errors in my editor or in the terminal.

I've removed the extra dependencies (they're included in the `eslint-config-standard-typescript` peer deps, so I'm not sure why they were added by the eslint init process...).

Also added concurrently to make the output from linting look a little nicer.

I don't think we can change the linter errors into warnings without making big changes to the config (it would mean going through every rule with `['error', ...]` and setting it to `['warn', ...]`). I've not been able to find a plugin that does this for the standard config.

Users can make their editor treat lint errors as warning though https://stackoverflow.com/a/69223673, we could suggest this setting in the readme in the section with the extensions.
